ckan/update-requirements.sh

```diff
@@ -8,12 +8,18 @@ cd requirements
 echo "Running poetry lock ..."
 poetry lock -vvv
 echo "Running poetry export ..."
-poetry export -vvv --format requirements.txt --output ../ckan/requirements.txt --without-hashes
-poetry export --dev -vvv --format requirements.txt --output ../ckan/requirements-dev.txt --without-hashes
+# Process to eliminate non-hashable sources #############################################
+poetry export -vvv --format requirements.txt --output ../ckan/requirements.tmp
+poetry export --dev -vvv --format requirements.txt --output ../ckan/requirements-dev.tmp
+# Now extract the packages that require editable (non-hashed) source from the hashable packages
+grep -v https ../ckan/requirements.tmp > ../ckan/requirements.txt
+grep -v https ../ckan/requirements-dev.tmp > ../ckan/requirements-dev.txt
+# Now extract the packages that require editable (non-hashed) source from the hashable packages
+grep https ../ckan/requirements-freeze.txt | grep -v httplib > ../ckan/requirements-noh.txt
```

Comment:
Two questions about this line: (1) why exclude httplib?  and (2) there is a bit of a chicken and egg problem here, since requirements-freeze.txt is also a derived file.  If I understand this code correctly, you'll be pulling an old requirements line from requirements-freeze.txt into requirements-now.txt.  Consider what would happen if we were trying to update the version of something that ends up in requirements-noh.txt.  If you could demonstrate/document how this would work, that's great.  But I'm not sure that it would.

Reply:
I have modified the update-requirements script to retrieve the correct commit hash for the appropriate editable source from the poetry.lock file.  I believe this should correct all the issues.

Reply:
So, not an unreasonable idea to pull the hash out of the poetry lock file! I do have a few concerns:
(1) Not clear to me that the poetry lock file has a guaranteed schema.  Do you know if the Poetry project has said that they have committed to that file format/contents?
(2) If we assume that is OK we read the file, the code in the last commit seems quite brittle (e.g., if Poetry adds a couple of keys to individual packages, I think the grep context size (the "-A3" arg) might need to be changed.  I'd feel *much* better if we were using code that was TOML aware to parse this file.  Probably lots of options, but I think the Python 'toml' library would work.